model-api-gen-gradle-test/build.gradle.kts

```diff
@@ -1,130 +1,16 @@
-    mavenCentral()
-    maven { url = uri("https://artifacts.itemis.cloud/repository/maven-mps/") }
-}
+import com.github.gradle.node.NodeExtension
+import com.github.gradle.node.NodePlugin
 
 plugins {
-    alias(libs.plugins.kotlin.jvm)
     base
-    id("org.modelix.model-api-gen")
-    alias(libs.plugins.node)
-}
-
-val mps by configurations.creating
-
-fun scriptFile(relativePath: String): File {
-    return file("$rootDir/build/$relativePath")
+    alias(libs.plugins.node) apply false
 }
 
-val mpsDir = buildDir.resolve("mps")
-
-val modelixCoreVersion: String = projectDir.resolve("../version.txt").readText()
-
-val kotlinGenDir = buildDir.resolve("metamodel/kotlin_gen")
-
-dependencies {
-    mps("com.jetbrains:mps:2021.1.4")
-    implementation("org.modelix:model-api-gen-runtime:$modelixCoreVersion")
-    implementation("org.modelix:modelql-typed:$modelixCoreVersion")
-    implementation("org.modelix:modelql-untyped:$modelixCoreVersion")
-
-    testImplementation(kotlin("test"))
-    testImplementation(kotlin("reflect"))
-
-    testImplementation("org.modelix:model-api:$modelixCoreVersion")
-    testImplementation("org.modelix:model-client:$modelixCoreVersion")
-    testImplementation("org.modelix:model-server-lib:$modelixCoreVersion")
-    testImplementation("org.modelix:modelql-client:$modelixCoreVersion")
-
-    testImplementation(libs.ktor.server.core)
-    testImplementation(libs.ktor.server.cors)
-    testImplementation(libs.ktor.server.netty)
-    testImplementation(libs.ktor.server.html.builder)
-    testImplementation(libs.ktor.server.auth)
-    testImplementation(libs.ktor.server.auth.jwt)
-    testImplementation(libs.ktor.server.status.pages)
-    testImplementation(libs.ktor.server.forwarded.header)
-    testImplementation(libs.ktor.server.websockets)
-    testImplementation(libs.ktor.server.test.host)
-    testImplementation(libs.logback.classic)
-
-    testImplementation(
-        files(kotlinGenDir) {
-            builtBy("generateMetaModelSources")
-        },
-    )
-}
-
-tasks.test {
-    useJUnitPlatform()
-}
-
-val resolveMps by tasks.registering(Sync::class) {
-    from(mps.resolve().map { zipTree(it) })
-    into(mpsDir)
-}
-
-sourceSets["main"].kotlin {
-    srcDir(kotlinGenDir)
-}
-
-metamodel {
-    mpsHeapSize = "2g"
-    dependsOn(resolveMps)
-    mpsHome = mpsDir
-    kotlinDir = kotlinGenDir
-    modelqlKotlinDir = kotlinGenDir
-    kotlinProject = project
-    typescriptDir = projectDir.resolve("typescript_src")
-    includeNamespace("jetbrains")
-    exportModules("jetbrains.mps.baseLanguage")
-
-    names {
-        languageClass.prefix = "L_"
-        languageClass.baseNameConversion = { it.replace(".", "_") }
-        typedNode.prefix = ""
-        typedNodeImpl.suffix = "Impl"
+subprojects {
+    plugins.withType<NodePlugin> {
+        project.extensions.configure<NodeExtension> {
+            version.set(libs.versions.node)
+            download.set(true)
+        }
```

Comment:
Is it necessary to configure the plugin here?
Isn't it just used in `/typescript-generation`?

Reply:
You are right. For now, it is only needed for `typescript-generation`.

But it will be needed for the coming `vue-generation`, so I already extracted it.